### PR TITLE
cpp_wrapper_cpu: Ensure reinterpret_view results in RAIIAtenTensorHandle

### PR DESCRIPTION
**Pull request by benjaminglass1**
Fixes segfaults caused by views being implicitly converted to AtenTensorHandle, then being destroyed before use.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140620
* #140817
* #140634
* #140624
* #139435
* #139412
* __->__ #139411

Closes #135559.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Comment by desertfire** [comment]
which specific test did this one fix? I can't remember it off the top of my head.

**Comment by benjaminglass1** [comment]
@desertfire I'd have to remind myself as well, but the error it fixed was a segfault resulting from an improper early resource destruction.

File: torch/_inductor/codegen/wrapper.py [hunk 1]
```diff
@@ -2059,14 +2059,21 @@ def make_buffer_reuse(self, old: BufferLike, new: BufferLike, delete_old: bool):
         reinterpret_view = self.codegen_reinterpret_view(
             old, new.get_size(), new.get_stride(), 0, self.wrapper_call
         )
-        return f"{self.declare_maybe_reference}{new_name} = {reinterpret_view}{del_line}  {self.comment} reuse"
+        move_begin = "std::move(" if V.graph.cpp_wrapper else ""
```
Comment (desertfire) [inline torch/_inductor/codegen/wrapper.py:2062]: We can make move_begin and move_end class members, similar to self.open_bracket and self.closed_bracket

**Review by desertfire — APPROVED** [review]

**Review by desertfire — COMMENTED** [review]
Please add a test by removing some skip_if_cpp_wrapper

File: torch/_inductor/codegen/wrapper.py (hunk 1 above)
Comment (benjaminglass1) [reply torch/_inductor/codegen/wrapper.py]: Done.

**Comment by benjaminglass1** [comment]
@kit1980 Do you have any details you can share on the internal test breakage? I'd like to get this fixed and re-merged fairly quickly.

**Comment by kit1980** [comment]
@desertfire could you help with re-landing this?
See D65833627 for the failures.

**Comment by Jack-Khuu** [comment]
Just checking in on the plans for this PR. 

This changes fixes an AOTI bug that we encountered when trying to bump the PT pin in https://github.com/pytorch/torchchat/pull/1367

**Comment by desertfire** [comment]
> @kit1980 Do you have any details you can share on the internal test breakage? I'd like to get this fixed and re-merged fairly quickly.

The test failure is about ArrayRef. @benjaminglass1 , let's just skip the changes to `torch/_inductor/codegen/cpp_wrapper_cpu_array_ref.py` in this PR.

**Pull request closed without merging**